main.go

```diff
@@ -114,35 +114,75 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Get non-flag command-line args
+	patterns := flag.Args()
+
 	// convert -skip flags to -ignore equivalents
 	for _, s := range skipExtensionFlags {
 		ignorePatterns = append(ignorePatterns, fmt.Sprintf("**/*.%s", s))
 	}
+
+	// create logger to print updates to stdout
+	logger := log.Default()
+
+	// real main
+	err := Run(
+		ignorePatterns,
+		spdx,
+		*holder,
+		*license,
+		*licensef,
+		*year,
+		*verbose,
+		*checkonly,
+		patterns,
+		logger,
+	)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// Run executes addLicense with supplied variables
+func Run(
+	ignorePatterns stringSlice,
+	spdx spdxFlag,
+	holder string,
+	license string,
+	licensef string,
+	year string,
```

Comment:
Combining stuff like holder, year, and license into a `licenseData` struct and passing that in would probably feel more appropriate.